### DISCUSSION TITLE
How would I use RSL to access data from the same tenant_id from a table?

### DISCUSSION THREAD

**Discussion by wilkensgomes** [supabase/supabase#15087] [Questions] [answered]
Do I have to create a function and then call it in a new policy? or do I just create a new policy? here's what I want more or less:

`CREATE OR REPLACE FUNCTION check_tenant_user_access() RETURNS BOOLEAN AS $$
BEGIN
    RETURN EXISTS (
        SELECT 1
        FROM tenant_user
        WHERE user_id = auth.uid()
    );
END;
$$ LANGUAGE plpgsql;`

**Comment by GaryAustin1** [comment]
Can you provide more context to your table and policies.   Not rally clear what you are trying to do without guessing.   If you are trying to do as select on the same table the policy is on for a select policy then you do need a function with security definer type to do that.

**Comment by wilkensgomes** [comment]
I have two tables the **tenant** table which contains _tenant_id_ and _name_ columns and I have the **tenant_user** table which contains _user_id_ and _tenant_id_ columns. And the third table **customer** and in that table there are columns _tenant_id_ _name_ and _phone_ for example. How do I make a Row Level Security so that users only have access to customers data belonging to their respective _tenant_id_?

**Reply by tholder** [reply, marked as answer]
You need to create a function to look up all the tenant_ids in the tenant_user table based on auth'd user, and then use that function in RLS.

Here is an example for my teams table:

```sql
CREATE OR REPLACE FUNCTION public.get_teams_for_authenticated_user()
 RETURNS SETOF uuid
 LANGUAGE sql
 STABLE SECURITY DEFINER
 SET search_path TO 'public'
AS $function$
    select team_id
    from team_members
    where user_id = auth.uid()
$function$
;
```

Then an RLS rule like:

```sql
CREATE POLICY "Select teams that authenticated user is member of" ON teams
    FOR SELECT
    USING (
        id IN (select get_teams_for_authenticated_user())
    );
```

**Reply by wilkensgomes** [reply]
Thank you very much. How would you add the team_id automatically when adding new data to the costumer table, for example?

**Reply by tholder** [reply]
You can do this with a trigger and another function with security definer.
